frappe/model/naming.py

```diff
@@ -322,25 +328,22 @@ def parse_naming_series(
 				digits = len(e)
 				part = number_generator(name, digits)
 				series_set = True
-		elif e == "YY":
-			part = today.strftime("%y")
-		elif e == "MM":
-			part = today.strftime("%m")
+		elif "YY" in e:
+			func, formatter = ("YYYY", "%Y") if len(e) > 2 and e[2] == "Y" else ("YY", "%y")
+			part = _get_formatted_date(e, func, formatter)
+		elif "MM" in e:
+			part = _get_formatted_date(e, "MM", "%m")
 		elif e == "DD":
 			part = today.strftime("%d")
-		elif e == "YYYY":
-			part = today.strftime("%Y")
 		elif e == "WW":
 			part = determine_consecutive_week_number(today)
 		elif e == "timestamp":
 			part = str(today)
 		elif e == "FY":
 			part = frappe.defaults.get_user_default("fiscal_year")
-		elif e.startswith("{") and doc:
+		elif doc and (e.startswith("{") or doc.get(e)):
 			e = e.replace("{", "").replace("}", "")
 			part = doc.get(e)
-		elif doc and doc.get(e):
-			part = doc.get(e)
```

Comment:
why is this removed?
we support fields in naming series with `prefix-.field.-####`

Reply:
it's combined into 1 condition - see L344